src/target/riscv/batch.c

```diff
@@ -249,3 +295,37 @@ bool riscv_batch_dmi_busy_encountered(const struct riscv_batch *batch)
 	const uint64_t in = buf_get_u64(field->in_value, 0, field->num_bits);
 	return get_field(in, DTM_DMI_OP) == DTM_DMI_OP_BUSY;
 }
+
+/* Returns the entry in `batch->idle_counts` that defines idle_count for the scan. */
+static idle_count_info_t *find_idle_counts_entry(struct riscv_batch *batch, size_t scan_idx)
+{
+	assert(!list_empty(&batch->idle_counts));
+
+	idle_count_info_t *entry;
+	list_for_each_entry(entry, &batch->idle_counts, list)
+		if (entry->until_scan > scan_idx)
+			break;
+	assert(!list_entry_is_head(entry, &batch->idle_counts, list));
+	return entry;
+}
```

Comment:
As noted above, this function becomes redundant when the list is removed.

Reply:
I've replied in [this comment](https://github.com/riscv-collab/riscv-openocd/pull/1028#discussion_r1560895289).